Drop support for old PHPUnit versions

greg0ire opened the pull request:
That implies bumping the version of PHP, and should make PHPUnit deprecations about `@before` and `@after` go away.

greg0ire committed:
Drop support for old PHPUnit versions

That implies bumping the version of PHP, and should make PHPUnit
deprecations about @before and @after go away.

greg0ire commented:
Technically not a breaking change, but given the amount of bumping I'm doing, I think it's best to bundle this with other breaking changes.

SenseException commented on .github/workflows/continuous-integration.yml:
```diff
@@ -27,5 +27,5 @@ jobs:
     uses: "doctrine/.github/.github/workflows/continuous-integration.yml@7.1.0"
     with:
-      php-versions: '["7.1", "7.2", "7.3", "7.4", "8.0", "8.1", "8.2", "8.3"]'
+      php-versions: '["8.1", "8.2", "8.3"]'
       composer-options: '--ignore-platform-req=php+'
```
Do we still need `--ignore-platform-req=php+` in this workflow? Not a blocker though.

greg0ire committed:
Remove flag that ignores the platform req

It was put in place to test with PHP 8.2 in fa6871b8abe23f2996e2809893c5e0f262f2dacd

driehle commented on .github/workflows/continuous-integration.yml:
```diff
@@ -27,5 +27,4 @@ jobs:
     with:
-      php-versions: '["7.1", "7.2", "7.3", "7.4", "8.0", "8.1", "8.2", "8.3"]'
-      composer-options: '--ignore-platform-req=php+'
+      php-versions: '["8.1", "8.2", "8.3"]'
```
Why not 8.4 here? It is allowed in composer.json.

greg0ire replied:
This MR was only about dropping support for old versions.
